### ISSUE TITLE
import-private-name raises for multi-root projects

### ISSUE THREAD

**Issue by kasium** [pylint-dev/pylint#10739]
### Bug description

```python
# p1/__init__.py
class _Dummy:
    pass

# p2/__init__.py
from p1 import _Dummy
```

### Configuration

```ini
[tool.pylint.master]
load-plugins = ["pylint.extensions.private_import",]
```

### Command used

```shell
pylint p1 p2
```

### Pylint output

```python
************* Module p2
p2/__init__.py(3): [C2701(import-private-name)] Imported private object (_Dummy)
```

### Expected behavior

If a project has multiple roots, the check should understand this. Currently it's implemented using path compares in `same_root_dir` but I think we need a configuration in which I can supply my package roots. If any of them is the start f the import, the import should never be flagged

### Pylint version

```shell
pylint 4.0.2
astroid 4.0.1
Python 3.14.0 (main, Nov 10 2025, 09:33:32) [GCC 7.5.0]
```

### OS / Environment

Linux

### Additional dependencies

```python

```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue.

Isn't it working as intended ? If the two projects are separated they should not import their prívate elements or it means they should in fact be in the same namespace or the element should not be private ?

**Comment by kasium** [comment]
I think it's debatable.

In my case the use case is detabale. The first project is a kind of SDK with a class hierarchy where the base class is private and child classes are exposed. So stakeholders of the SDK should not use the base class. In the same folder structure and deliverable we have the second project which uses the SDK and as its in our hands, we are allowed to use the base class. Hope that makes somehow sense 😅
